docs/source/developers/python.rst

```diff
@@ -586,6 +586,86 @@ Caveats
 
 The Plasma component is not supported on Windows.
 
+Relevant components and environment variables
+=============================================
+
+List of relevant Arrow CMake flags and corresponding environment variables
+to be used when building PyArrow are:
+
+.. list-table::
+   :widths: 30 30
+   :header-rows: 1
+
+   * - CMAKE_BUILD_TYPE
+     - PYARROW_BUILD_TYPE (release, dbug or relwithdebinfo)
+   * - ARROW_GCS
+     - PYARROW_WITH_GCS
+   * - ARROW_S3
+     - PYARROW_WITH_S3
+   * - ARROW_HDFS
+     - PYARROW_WITH_HDFS
+   * - ARROW_CUDA
+     - PYARROW_WITH_CUDA
+   * - ARROW_SUBSTRAIT
+     - PYARROW_WITH_SUBSTRAIT
+   * - ARROW_FLIGHT
+     - PYARROW_WITH_FLIGHT
+   * - ARROW_DATASET
+     - PYARROW_WITH_DATASET
+   * - ARROW_PARQUET
+     - PYARROW_WITH_PARQUET
+   * - PARQUET_REQUIRE_ENCRYPTION
+     - PYARROW_WITH_PARQUET_ENCRYPTION
+   * - ARROW_PLASMA
+     - PYARROW_WITH_PLASMA
+   * - ARROW_TENSORFLOW
+     - PYARROW_WITH_TENSORFLOW
+   * - ARROW_ORC
+     - PYARROW_WITH_ORC
+   * - ARROW_GANDIVA
+     - PYARROW_WITH_GANDIVA
+
+List of relevant environment variables that can also be used to build
+PyArrow are:
+
+.. list-table::
+   :widths: 30 30
+   :header-rows: 1
+
+   * - PyArrow environment variable
+     - Description
+   * - PYARROW_CMAKE_GENERATOR
+     - Example: 'Visual Studio 15 2017 Win64'
+   * - PYARROW_CMAKE_OPTIONS
+     - Extra CMake and Arrow options (ex. ``"-DARROW_SIMD_LEVEL"``,
+       ``"-DCMAKE_OSX_ARCHITECTURES"``)
```

Comment:
* How about specify option value explicitly in example because users need to specify option value?
* How about specify two options in example to show users can specify multiple options?

```suggestion
     - Extra CMake and Arrow options (ex. ``"-DARROW_SIMD_LEVEL=NONE -DCMAKE_OSX_ARCHITECTURES=x86_64;arm64"``)
```